### PR TITLE
[HDInsight] Replace the referenced HDInsight .NET SDK from Microsoft. Azure. Management. HDInsight to Azure. ResourceManager. HDInsight

### PR DESCRIPTION
**Pull request by yuc-Li**
## Description

- Replace the referenced HDInsight .NET SDK from `Microsoft.Azure.Management.HDInsight` to `Azure.ResourceManager.HDInsight`(Here are some break changes).
- New feature: added parameter `EnableSecureChannel` enable secure channel when create a cluster.
- Fix bug: fixed cannot get storage account info when use gen2 type.

Fewer code changes for bug fix and new features, so opening more PR will not reduce the number of changed files.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/HDInsight/HDInsight/ManagementCommands/RemoveAzureHDInsightClusterCommand.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public override void ExecuteCmdlet()
                 ResourceGroupName = GetResourceGroupByAccountName(ClusterName);
             }
 
-            HDInsightManagementClient.DeleteCluster(ResourceGroupName, ClusterName);
+            HDInsightManagementClient.DeleteClusterAsync(ResourceGroupName, ClusterName);
```
Comment (aim-for-better) [inline src/HDInsight/HDInsight/ManagementCommands/RemoveAzureHDInsightClusterCommand.cs:48]: please use the sync method to keep the same behavior with previous

Comment (yuc-Li) [reply src/HDInsight/HDInsight/ManagementCommands/RemoveAzureHDInsightClusterCommand.cs]: The behavior is the same as before, and the two names have been changed back

**Comment by YanaXu** [comment]
Hi @yuc-Li, there are breaking changes in this PR. Could you hold this PR to the next major release and now add the breaking changes warnings?

**Comment by yuc-Li** [comment]
> 

Sure, I will first submit a PR to add warning messages.

**Comment by yuc-Li** [comment]
Hi @YanaXu, please review this PR, we plan to complete it in the upcoming major release.

**Pull request closed without merging**